backend/experiment/serializers.py

```diff
@@ -3,14 +3,138 @@
 
 from django_markup.markup import formatter
 from django.utils.translation import activate, get_language
+from rest_framework import serializers
 
 from experiment.actions.consent import Consent
 from image.serializers import serialize_image
 from participant.models import Participant
 from result.models import Result
 from session.models import Session
 from theme.serializers import serialize_theme
-from .models import Block, Experiment, Phase, SocialMediaConfig
+from .models import Block, Experiment, Phase, SocialMediaConfig, ExperimentTranslatedContent
+
+
+class ExperimentTranslatedContentSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = ExperimentTranslatedContent
+        fields = ["id", "index", "language", "name", "description", "about_content", "social_media_message"]
+
+
+class BlockSerializer(serializers.ModelSerializer):
+    id = serializers.IntegerField(required=False)
+
+    class Meta:
+        model = Block
+        fields = [
+            "id",
+            "index",
+            "slug",
+            "rounds",
+            "bonus_points",
+            "rules",
+        ]
+        extra_kwargs = {
+            "slug": {"validators": []},
+        }
+
+
+class PhaseSerializer(serializers.ModelSerializer):
+    blocks = BlockSerializer(many=True, required=False)
+
+    class Meta:
+        model = Phase
+        fields = ["id", "index", "dashboard", "randomize", "blocks"]
+
+
+class ExperimentSerializer(serializers.ModelSerializer):
```

Comment:
This is the weak part about this setup (for now) I think. It seems to be a bit convoluted but I do think it should be able to simplify it. It's like this right now because I had to deal with different scenarios like many to many relationships, or just FKs, do the related objects have PKs already or not, and so on. Maybe a good step is just to create and save an empty experiment before you get to the form. Then at least the experiment already exists and we can remove the `create` method in here.